### ISSUE TITLE
Nette does not receive array

### ISSUE THREAD

**Issue by petrparolek** [jiripudil/Naja#202]
Naja version: 2.1.2

JS:

```
            year = '2020';
            paymentTags = {1,2};

            data = { year, paymentTags };

            naja.makeRequest('GET', "/?do=paymentsJS-changeYear", data)
```
![obrazek](https://user-images.githubusercontent.com/6066243/104620443-4c93d180-568f-11eb-812e-7b367d61297c.png)

Nette receives

![obrazek](https://user-images.githubusercontent.com/6066243/104620225-0b9bbd00-568f-11eb-8154-df3f4a38be15.png)

**Comment by jiripudil** [comment]
Reposting my thoughts from #207 for reference:

> Since this issue is already on the table, I'd like to have it fixed properly, i.e.
>
> - Naja should transform everything that PHP and/or Nette can correctly parse, including arrays (`key[]=value`), objects (`key[subkey]=value`), and any arbitrarily nested structures (`key[subkey][0][subkey]=value`).
> - In addition to GET and HEAD requests where data need to be serialized into the query string, the same transformation also needs to be done to `application/x-www-form-urlencoded` request body – currently it suffers from pretty much the same bug [on this line](https://github.com/jiripudil/Naja/blob/master/src/Naja.ts#L116).
> - Needless to say, this kind of change needs a decent test coverage.

**Comment by petrparolek** [comment]
Based on last posts about Naja I invent elegant solution of my problem without any changes in Naja :

```
            data = { year, paymentTags };
            var dataString = JSON.stringify(data);
            console.log(dataString);

            naja.makeRequest('GET', {link changeYear!}, { paramsJson: dataString}, { dataType: 'json'})
````

and in PHP:
```
	public function handleChangeYear(): void
	{
		if ($this->presenter->isAjax()) {
			$paramsJson =$this->presenter->getParameter('paramsJson');
			$params = Json::decode($paramsJson);
			bdump($params);
			$year = (int) $params->year;
			bdump($year);
```

**Comment by jiripudil** [comment]
I've released the fix in [2.1.3](https://github.com/jiripudil/Naja/releases/tag/2.1.3)
